src/ripple/shamap/impl/SHAMapSync.cpp

```diff
@@ -190,11 +193,13 @@ SHAMap::getMissingNodes(std::vector<SHAMapNodeID>& nodeIDs, std::vector<uint256>
                         {
                             if (!pending)
                             { // node is not in the database
-                                nodeIDs.push_back (childID);
-                                hashes.push_back (childHash.as_uint256());
+                                ret.push_back (
+                                    std::make_pair (
+                                        childID,
+                                        childHash.as_uint256()));
```

Comment:
I think using `emplace_back` instead of `push_back(max_pair` will save an allocate/copy/free. It's cleaner even if it doesn't.